sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/TokenValidatorInternal.cs

```diff
@@ -1,127 +1,77 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents;
-using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework;
-using Microsoft.IdentityModel.Tokens;
 using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.IdentityModel.Tokens.Jwt;
 using System.Linq;
 using System.Net.Http;
-using System.Security.Cryptography;
+using System.Security.Claims;
 using System.Threading;
 using System.Threading.Tasks;
 
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Logging;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
 namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
 {
     internal class TokenValidatorInternal : TokenValidator
     {
-        internal static List<SecurityKey> _cacheKeys = new List<SecurityKey>();
-        internal static DateTime? _lastSync;
-        internal static TimeSpan _cacheRefresh = new TimeSpan(0, 0, 5, 0);
-        private static readonly SemaphoreSlim _semaphore = new SemaphoreSlim(1);
-
-        internal override async Task<(bool Valid, Dictionary<string, string> Claims)> GetClaimsAndValidate(HttpRequestMessage request, ConfigurationManager configurationManager)
+        internal async override Task<Dictionary<string, string>> ValidateAndGetClaims(
+            HttpRequestMessage request,
+            ConfigurationManager configurationManager)
         {
-            if (string.IsNullOrEmpty(configurationManager.TenantId) || string.IsNullOrEmpty(configurationManager.AudienceAppId))
-            {
-                throw new MissingFieldException(string.Format(CultureInfo.CurrentCulture, AuthenticationEventResource.Ex_Trigger_Required_Attrs, ConfigurationManager.TENANT_ID, ConfigurationManager.AUDIENCE_APPID));
-            }
+            IdentityModelEventSource.ShowPII = ConfigurationManager.ShowPIIDataInLogs;
 
             string accessToken = request.Headers?.Authorization?.Parameter;
 
             if (string.IsNullOrWhiteSpace(accessToken))
             {
-                return (false, null);
+                throw new UnauthorizedAccessException(AuthenticationEventResource.Ex_Invalid_Token);
```

Comment:
This seems like an odd exception type for a validation method. Should it be something like `InvalidOperationException`?

Reply:
In `AuthenticationEventBinding.GetFaultyRequest`, we look for `UnauthorizedAccessException` to see if it's a token access error.

Any Access Token Validation and exception should be `UnauthorizedAccessException` so we can reply with the correct to response.